src/material/schematics/ng-generate/mdc-migration/rules/theming-styles.ts

```diff
@@ -45,9 +45,7 @@ export class ThemingStylesMigration extends Migration<ComponentMigrator[], Schem
     if (migrator) {
       migrator.styles.replaceMixin(this.namespace, atRule);
     } else if (atRule.params.includes('all-component-themes') && atRule.parent) {
-      this.upgradeData.forEach(m => {
-        m?.styles.addNewMixinsAfterNode(this.namespace, atRule);
-      });
+      this.upgradeData[0]?.styles.replaceAllComponentThemeMixin(atRule);
```

Comment:
Nit: should we add a comment saying that we can pick any arbitrary migrator here since this replacement is not specific to a certain component? (a little magical but should be fine)